### PR TITLE
Gaydardz/notifications custom icons develop

### PR DESCRIPTION
**Pull request by DimitarGaydardzhiev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -231,6 +231,8 @@ Take a look at the following overview of the Admin App extension samples we prov
 
 * [Custom list components](./src/tree#custom-list-components) - This extension is used to replace a part of each item shown in the tree used throughout the AdminApp. Currently supported only for the related data functionality.
 
+* [Custom notifications icons](./src/custom-notifications-icons) - This extension is used to generate css class that can be used to style the notification icon. If you have some custom notifications you can use this extension to create your own icon, isntead of using the default one. You can also use it to override the already available icons in the system.
```
Comment (Novkirishki) [inline README.md:234]: Lets make it clear that this is about system notifications and not app notifications (like the one that appears when you publish, etc.). Maybe add a link to the SF documentation as well

File: src/custom-notifications-icons/index.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * The notifications icons extender module.
+ */
+@NgModule({
+    providers: [
+        CUSTOM_NOTIFICATION_ICON_PROVIDER
+    ]
+})
+export class NotificationsIconsExtenderModule { /* empty */ }
```
Comment (Novkirishki) [inline src/custom-notifications-icons/index.ts:12]: add system keyword everywhere

File: src/custom-notifications-icons/index.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from "@angular/core";
+import { CUSTOM_NOTIFICATION_ICON_PROVIDER } from "./custom-notification-icon-provider";
+
+/**
+ * The notifications icons extender module.
+ */
+@NgModule({
+    providers: [
+        CUSTOM_NOTIFICATION_ICON_PROVIDER
```
Comment (Novkirishki) [inline src/custom-notifications-icons/index.ts:9]: add system keyword

File: src/custom-notifications-icons/README.MD [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Custom notification icon
+
+## Implementation
+
+Each system notification coming from Sitefinity has a property **NotificationKey**. This property is used to build the class of the icon for the specific notification. Whether you create your own custom notifications, or you are working with the ones already registered in Sitefinity, you can interact with this property to change the icon of the notification.
```
Comment (Novkirishki) [inline src/custom-notifications-icons/README.MD:5]: "interact" does not seem appropriate here

File: src/custom-notifications-icons/README.MD [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+				case "subscribed-responses":
+					return NEW_FORM_RESPONSE_ICON;
+				// override the default icon
+				default:
+					return DEFAULT_ICON;
+			}
+		}
+	}
+
+In the above example, the key argument, passed to the `parseIcon()` function is the **NotificationKey**, coming from Sitefinity. Here are covered the following scenarios:
```
Comment (Novkirishki) [inline src/custom-notifications-icons/README.MD:35]: maybe we should expose the default notifications keys as enum or consts

File: src/custom-notifications-icons/custom-notification-icon-provider.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { ClassProvider, Injectable } from "@angular/core";
+import { SystemNotificationIconProvider, SYSTEM_NOTIFICATION_ICON_TOKEN } from "@progress/sitefinity-adminapp-sdk/app/api/v1";
+
+const DEFAULT_ICON = "check";
```
Comment (Novkirishki) [inline src/custom-notifications-icons/custom-notification-icon-provider.ts:4]: are these font awesome icons? lets explain this to clients

File: README.md (hunk 1 above)
Comment (DimitarGaydardzhiev) [reply README.md]: We don't have a documentation link (it is not completed yet), but will log an action item to add it, when it is ready

File: README.md [hunk 7]
```diff
@@ -231,6 +231,8 @@ Take a look at the following overview of the Admin App extension samples we prov
 
 * [Custom list components](./src/tree#custom-list-components) - This extension is used to replace a part of each item shown in the tree used throughout the AdminApp. Currently supported only for the related data functionality.
 
+* [Custom system notifications icons](./src/custom-system-notifications-icons) - This extension is used to generate css class that can be used to style a system notification icon. If you have some custom system notifications you can use this extension to create your own icon, instead of using the default one. You can also use it to override the already available icons in the system.
```
Comment (Novkirishki) [inline README.md:234]: to **provide** your own icon

**Review by Novkirishki — APPROVED** [review]

**Review by FerdiNebi — APPROVED** [review]

**Pull request merged**